aws/resource_aws_budgets_budget.go

```diff
@@ -0,0 +1,431 @@
+						Default:  true,
+					},
+					"use_blended": {
+						Type:     schema.TypeBool,
+						Optional: true,
+						Default:  false,
+					},
+				},
+			},
+		},
+		"time_period_start": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		"time_period_end": {
+			Type:     schema.TypeString,
+			Optional: true,
+			Default:  "2087-06-15_00:00",
+		},
+		"time_unit": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		"cost_filters": {
+			Type:     schema.TypeMap,
+			Optional: true,
+			Computed: true,
+		},
+	}
+}
+
+func resourceAwsBudgetsBudgetCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AWSClient).budgetconn
+	var accountID string
+	if v, ok := d.GetOk("account_id"); ok {
+		accountID = v.(string)
+	} else {
+		accountID = meta.(*AWSClient).accountid
+	}
+
+	budget, err := expandBudgetsBudgetUnmarshal(d)
+	if err != nil {
+		return fmt.Errorf("failed creating budget: %v", err)
+	}
+
+	_, err = client.CreateBudget(&budgets.CreateBudgetInput{
+		AccountId: &accountID,
+		Budget:    budget,
+	})
+	if err != nil {
+		return fmt.Errorf("create budget failed: %v", err)
+	}
+
+	d.SetId(fmt.Sprintf("%s:%s", accountID, *budget.BudgetName))
+	return resourceAwsBudgetsBudgetUpdate(d, meta)
+}
+
+func resourceAwsBudgetsBudgetRead(d *schema.ResourceData, meta interface{}) error {
+	accountID, budgetName, err := decodeBudgetsBudgetID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	client := meta.(*AWSClient).budgetconn
+	describeBudgetOutput, err := client.DescribeBudget(&budgets.DescribeBudgetInput{
+		BudgetName: &budgetName,
+		AccountId:  &accountID,
+	})
+	if isAWSErr(err, budgets.ErrCodeNotFoundException, "") {
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("describe budget failed: %v", err)
+	}
+
+	flattenedBudget, err := expandBudgetsBudgetFlatten(describeBudgetOutput.Budget)
+	if err != nil {
+		return fmt.Errorf("failed flattening budget output: %v", err)
+	}
+
+	if _, ok := d.GetOk("name"); ok {
+		d.Set("name", flattenedBudget.name)
+	}
+
+	for k, v := range map[string]interface{}{
+		"account_id":        accountID,
+		"budget_type":       flattenedBudget.budgetType,
+		"time_unit":         flattenedBudget.timeUnit,
+		"cost_filters":      convertCostFiltersToStringMap(flattenedBudget.costFilters),
+		"limit_amount":      flattenedBudget.limitAmount,
+		"limit_unit":        flattenedBudget.limitUnit,
+		"cost_types":        []interface{}{flattenedBudget.costTypes},
+		"time_period_start": flattenedBudget.timePeriodStart.Format("2006-01-02_15:04"),
+		"time_period_end":   flattenedBudget.timePeriodEnd.Format("2006-01-02_15:04"),
+	} {
+		if _, ok := d.GetOk(k); ok {
+			if err := d.Set(k, v); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func resourceAwsBudgetsBudgetUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AWSClient).budgetconn
+	var accountID string
+	if v, ok := d.GetOk("account_id"); ok {
```

Comment:
If we don't need to call update after create (its not typical), this logic gets replaced by the `decodeBudgetsBudgetId()` call 👍

Reply:
👍